docs/src/pages/index.mdx

```diff
@@ -0,0 +1,178 @@
+//    ^? [string, string, string]
+```
+
+## 🦾 Functional
+
+Functional programming is beneficial, making code more readable. However, there
+are situations where you don't need "pipes" and want to call just a single
+function. In Remeda, **all** functions have automatic currying built-in. This
+means the same imported function can be called either regularly (what we call
+_dataFirst_) or with the first parameter, _data_, omitted (what we call
+_dataLast_). When a dataLast invocation is detected, Remeda creates a curried
+version of that function on-the-fly, with all other parameters bound to it. This
+allows you to use the same function in both functional and imperative styles,
+without additional imports. It also means you can gradually adopt a functional
+style where it makes sense without compromising readability when it doesn't.
+
+```ts
+// Lodash
+_.pick(obj, ["firstName", "lastName"]);
+
+// Ramda
+R.pick(["firstName", "lastName"], obj);
+
+// In Remeda you get both!
+R.pick(obj, ["firstName", "lastName"]); // dataFirst
+R.pipe(obj, R.pick(["firstName", "lastName"])); // dataLast
+```
+
+In the above example, the "dataFirst" approach is more natural and programmer-
+friendly because when you type the second argument, you get auto-complete from
+the IDE. It's not possible to get auto-complete in Ramda because the data
+argument is not provided.
+
+The "dataLast" approach is helpful when writing data transformations, aka pipes.
+
+```ts
+const DATA = [
+  { name: "john", age: 20, gender: "m" },
+  { name: "marry", age: 22, gender: "f" },
+  { name: "samara", age: 24, gender: "f" },
+  { name: "paula", age: 24, gender: "f" },
+  { name: "bill", age: 33, gender: "m" },
+];
+
+// Remeda
+R.pipe(
+  DATA,
+  R.filter((x) => x.gender === "f"),
+  R.groupBy((x) => x.age),
+);
+
+// Ramda
+R.pipe(
+  R.filter((x) => x.gender === "f"),
+  R.groupBy((x) => x.age),
+)(DATA); // broken typings in TS :(
+
+// Lodash
+_(DATA)
+  .filter((x) => x.gender === "f")
+  .groupBy((x) => x.age)
+  .value();
+
+// Lodash-fp
+_.flow(
+  _.filter((x) => x.gender === "f"),
+  _.groupBy((x) => x.age),
+)(DATA); // broken typings in TS :(
+```
+
+## 🔧 Lazy Evaluation
+
+You might have noticed the [`pipe`](/docs#pipe) function in the above example.
+The pipe function allows you to structure complex data transformations in the
+order they will be computed, avoiding the need to name intermediate variables or
+having confusingly deep nesting that needs to be read from the center outwards.
+
+But the Remeda pipe function has an additional benefit. It automatically detects
+Remeda functions that could be computed **lazily** and optimizes those parts of
+the pipe without you having to do anything. This means that expensive
+computations can be deferred until they are actually needed.
+
+```ts
+// Get first 3 unique values
+const DATA = [1, 2, 2, 3, 3, 4, 5, 6];
+
+const result = R.pipe(
+  DATA,
+  R.map((x) => {
+    console.log("iterate", x);
+    return x;
+  }),
```

Comment:
**nit:** replace with `R.tap`? it's a bit less readable though, hm

Reply:
It's less understandable as an example, and causes prettier to put everything in a single line which makes it less readable too, I think that as an example this is clearer